### PR TITLE
feat: add product mix component

### LINKED ISSUE #195: Product Mix List
Goal:
As a system admin, I want to view all product mix, as seen here
https://demo.openmf.org/#/productmix
9_productmixlist

Resources:
https://demo.openmf.org/api-docs/apiLive.htm#loanproductmix
https://mifosforge.jira.com/wiki/spaces/docs/pages/103186467/Product+Mix

### PR DESCRIPTION
**Pull request by jonathanzhang53**
## Description
Implements product mix component

## Related issues and discussion
#195

## GIF
mixree

## Checklist
Please make sure these boxes are checked before submitting your pull request - thanks!

- [x] If you have multiple commits please combine them into one commit by squashing them.

- [x] Read and understood the contribution guidelines at `web-app/.github/CONTRIBUTING.md`.

### CODE REVIEW

**Comment by jonathanzhang53** [comment]
Done @abhaychawla Thank you for reviewing my PRs. Look forward to working further with you on this repo in the future.

File: src/app/products/products.service.ts [hunk 1]
```diff
@@ -42,4 +42,16 @@ export class ProductsService {
     return this.http.get('/fixeddepositproducts');
   }
 
+  /**
+   * @param {any} Fetches Product mixes.
+   * @returns {Observable<any>}
+   */
+  getProductMixes(useHttpParams: boolean = false): Observable<any> {
+    let httpParams = new HttpParams();
+    if (useHttpParams) {
```
Comment (abhaychawla) [inline src/app/products/products.service.ts]: Remove this, we will always be having http params here as this is specifically for getting product mix data.

**Comment by jonathanzhang53** [comment]
done

File: src/app/products/products.service.ts [hunk 2]
```diff
@@ -42,4 +42,14 @@ export class ProductsService {
     return this.http.get('/fixeddepositproducts');
   }
 
+  /**
+   * @param {any} Fetches Product mixes.
```
Comment (abhaychawla) [inline src/app/products/products.service.ts]: There is no parameter passed here. Remove this.

**Comment by jonathanzhang53** [comment]
I fixed it for the rest of the service functions I wrote.

>

**Comment by abhaychawla** [comment]
My bad I didn't see that earlier. Don't fix for the other functions in this PR. We'll have to take care of that separately. Just fix for the function used for this PR.

**Pull request merged**